dbt/adapters/athena/session.py

```diff
@@ -261,9 +298,13 @@ def release_session_lock(self, session_id: str) -> None:
         if session_status["State"] in ["IDLE", "BUSY"] and (
             session_status["StartDateTime"] - datetime.now(tz=timezone.utc) > timedelta(seconds=self.timeout)
         ):
-            LOGGER.debug(f"Terminating session: {session_id}")
-            self.athena_client.terminate_session(SessionId=session_id)
-        with self.lock:
+            LOGGER.debug(f"Terminating timed out session: {session_id}")
+            self.terminate_session(session_id)
+
+        if session_status["State"] in ["TERMINATING", "TERMINATED", "DEGRADED", "FAILED"]:
+            spark_session_locks.pop(UUID(session_id), "Session id not found")
+
+        with self.lock:  # why do we need this context manager? Why cant it be released without this?
```

Comment:
@Avinash-1394  do we need this context manager? _I am not an expert with this threading library_ :-)